connector/dialect.tdd

```diff
@@ -1,375 +1,179 @@
-	ELSE %2 END)</formula>
-      <argument type='datetime' />
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='MAX' return-type='date'>
-      <formula>(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 &gt; %2 THEN %1
-	ELSE %2 END)</formula>
-      <argument type='date' />
-      <argument type='date' />
-    </function>
-    <function group='date' name='MIN' return-type='datetime'>
-      <formula><![CDATA[(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 < %2 THEN %1
-	ELSE %2 END)]]></formula>
-      <argument type='datetime' />
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='MIN' return-type='date'>
-      <formula><![CDATA[(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 < %2 THEN %1
-	ELSE %2 END)]]></formula>
-      <argument type='date' />
-      <argument type='date' />
-    </function>
-    <function group='date' name='MONTH' return-type='int'>
-      <formula>CAST(EXTRACT(MONTH FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='NOW' return-type='datetime'>
-      <formula>CURRENT_TIMESTAMP</formula>
-    </function>
-    <function group='date' name='QUARTER' return-type='int'>
-      <formula>CAST(EXTRACT(QUARTER FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='TODAY' return-type='date'>
-      <formula>CURRENT_DATE</formula>
-    </function>
-    <function group='date' name='WEEK' return-type='int'>
-      <formula>CAST(FLOOR((7 + EXTRACT(DOY FROM %1) - 1 + EXTRACT(DOW FROM DATE_TRUNC(&apos;YEAR&apos;, %1))) / 7) AS INTEGER)</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='YEAR' return-type='int'>
-      <formula>CAST(EXTRACT(YEAR FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>date_trunc(&apos;day&apos;, CAST(&apos;1900-01-01&apos; AS DATE) + %1 * INTERVAL &apos;1 DAY&apos;)</formula>
+    <function group='cast' name='BOOL' return-type='bool'>
+      <formula>CAST(%1 AS BOOL)</formula>
       <argument type='real' />
     </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>CAST(%1 AS DATE)</formula>
-      <argument type='str' />
-    </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>CAST(%1 AS DATE)</formula>
-      <argument type='datetime' />
+    <function group='cast' name='BOOL' return-type='bool'>
+      <formula>CAST(%1 AS BOOL)</formula>
+      <argument type='int' />
     </function>
-    <function group='cast' name='DATETIME' return-type='datetime'>
-      <formula>(CAST(&apos;1900-01-01 00:00:00&apos; AS TIMESTAMP) + %1 * INTERVAL &apos;1 DAY&apos;)</formula>
-      <argument type='real' />
+    <function group='cast' name='BOOL' return-type='bool'>
```

Comment:
Maybe if you have time, can you test this out: Instead of CAST(%1 AS BOOL), what about just using %1 ? Because the cast is not necessary when cast to the same type.

Reply:
I removed the cast where the argument type matched the type being casted to, STR, INT, FLOAT, DATETIME, BOOL. 
I ran a quick test for `BOOL` and that returned the boolean type as expected